DataFormats/Portable/interface/PortableCollectionCommon.h

```diff
@@ -0,0 +1,113 @@
+
+#ifndef DataFormats_Portable_interface_PortableCollectionCommon_h
+#define DataFormats_Portable_interface_PortableCollectionCommon_h
+
+template <auto Start, auto End, auto Inc, class F>
+constexpr void constexpr_for(F&& f) {
+  if constexpr (Start < End) {
+    f(std::integral_constant<decltype(Start), Start>());
+    constexpr_for<Start + Inc, End, Inc>(f);
+  }
+}
+
+template <std::size_t idx, typename T>
```

Comment:
can you change `idx` to `Idx` ?
```suggestion
template <std::size_t Idx, typename T>
```